src/Symfony/Component/Serializer/Mapping/Factory/ClassMetadataFactory.php

```diff
@@ -115,4 +115,9 @@ private function getClass($value)
 
         return ltrim(is_object($value) ? get_class($value) : $value, '\\');
     }
+
+    protected function createMetadataClass($class)
```

Comment:
also can you move https://github.com/EXSyst/symfony/blob/2.7/src/Symfony/Component/Serializer/Mapping/Factory/ClassMetadataFactory.php#L67-L69 into this method ?